entities.c

```diff
@@ -390,3 +390,32 @@ size_t decode_html_entities_utf8(char *dest, const char *src)
 	return (size_t)(to - dest);
 }
 
+int encode_html_entities(char *dest, const char *src) {
+        char *to = dest;
+        for( const char *from = src ; *from ; from++ ) {
+            int i = 9999;
+            if ( *from <= '+' ) {
+                sprintf(to,"%%%02x",*from);
+                to += 3;
+                continue;
+            }
+            if ( *from<='z' ) {
+                *to = *from;
+                to++;
+                continue;
+            }
+            //if ( *from=='\r' || *from=='\n' ) continue;
+            for( i=0 ; i<sizeof NAMED_ENTITIES / sizeof *NAMED_ENTITIES ; i++ )
+                if ( *from == NAMED_ENTITIES[i][1][0] ) break;
```

Comment:
I don't really get this logic :-) are you comparing just the first character or am I missing something? Or is this sufficient because of the available entities (then please make sure this remains the case by adding an appropriate test case)

Reply:
Yes, I'm just comparing the first character. I found this sufficient for the final application I was developing. Maybe this encoder should be classified as an ASCII encoder instead of an UTF-8 encoder.  I'll think about this during next week. Enlarging the length of the test can be done but this solution will still be inefficient since it uses a top-down search on an unsorted table.
If I won't improve the test, I'll try to keep the restriction clear to future users or developers.
Any way, I'll provide a test case.